impl/src/main/java/com/okta/sdk/impl/util/OAuth2Utils.java

```diff
@@ -0,0 +1,163 @@
+package com.okta.sdk.impl.util;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.okta.commons.http.*;
+import com.okta.commons.http.okhttp.OkHttpRequestExecutorFactory;
+import com.okta.sdk.impl.config.ClientConfiguration;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemReader;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+
+public class OAuth2Utils {
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    private static List<String> allowedSigningAlgorithms =
+        new ArrayList<>(Arrays.asList("RS256", "RS384", "RS512", "ES256", "ES384", "ES512"));
+
+    private static final OkHttpRequestExecutorFactory okHttpRequestExecutorFactory =
+        new OkHttpRequestExecutorFactory();
+
+    private static KeyFactory getKeyFactoryInstance() throws NoSuchAlgorithmException {
+        return KeyFactory.getInstance("RSA");
```

Comment:
This hardcodes `RSA` here.  JJWT, should provide libraries to manage all of this for you IIRC (other than the PEM parsing)

Reply:
Will update code to try pem parsing with RSA keyfactory by default first. If it fails, we will check for EC next. If both fail, then InvalidKeySpecException will be thrown.